doc/admin-guide/files/records.yaml.en.rst

```diff
@@ -3559,17 +3559,25 @@ URL Remap Rules
 
    This is dynamic to enable different requirements for startup and reloading.
 
-.. ts:cv:: CONFIG proxy.config.url_remap.acl_matching_policy INT 0
+.. ts:cv:: CONFIG proxy.config.url_remap.acl_behavior_policy INT 0
    :reloadable:
 
-   This controls matching policy of ACL filters in :file:`remap.config`. See :ref:`acl-filters` for more details.
+   This controls how ACL filter actions behave in :file:`remap.config`. See :ref:`acl-filters` for more details.
 
-   ===== =============================
+   ===== =================================
    Value Description
-   ===== =============================
-   ``0`` Match on IP and Method Policy
-   ``1`` Match on IP only Policy
-   ===== =============================
+   ===== =================================
+   ``0`` Legacy (9.x and below) behavior.
+   ``1`` Modern (10.x and above) behavior.
```

Comment:
I still don't have motivation to rename them to `Legacy` and `Modern`. I don't see why we can't keep it as is which describes the behavior well.

Reply:
Thank you for sharing this. It's good to discuss this.

Here's my logic: in changing the names I'm aiming for us to communicate to the end user what I think is the most important aspect of this configuration: which is the "old" compatibility mode and which is the "new" incompatibility mode. In other words, for most users what will be the most crucial aspect is "Which configuration will allow me to use my current remap.config without breaking things?" I chose the names "legacy" and "modern" because those words seem more noble than "old" versus "new". :) But I'm open to other suggestions.

My hesitation with "Match on IP and Method" and "Match on IP Only" is that, while they are literally accurate, aside from you, Matty, and me who have worked very closely with this system recently, I'm guessing those names will not likely be readily obvious in their meaning to most users. If we presented those names to a random ops person at Apple, do you think they'll know which one is the 9.x behavior confidently? They might be able to reason through it, but it might add a bit of a cognitive load. By the way, I'm not blaming you for these names, I did after all collaborate with you on them.

Having said that, I might be wrong. This really isn't a big deal to me. I'm glad you added the configuration in the first place, that will really help this release. Maybe the original names will help users consider the actual behaviors - that's their purpose. Either way we'll clearly document which is the compatibility mode and which is the new, which is default, and what our 11.x plans are.

So I'll leave it up to you. If after reading the above, you still want the original names, I'll happily leave it. Or if you have other naming suggestions, I'll consider that too.

Thank you for the review.

Reply:
Thanks for explanation. I got your point.

One thing I'd like to clarify, sorry for repeating what I pointed out on ML, is that "Match on IP and Method Policy" is pretty similar to the behavior of 9.2.x and older, so almost all of the old configs will work as is. However, it employs "first match win" policy between ACL filters in remap.config and ip_allow.yaml. Then, strictly speaking, we can't say it's 100% compatible with old configs. I'd still say calling it "Legacy" is a kind of mis-leading. I wrote this in the "Note" of the Match on IP and Method Policy docs. I welcome to make it more clear, but I prefer to keep calling it "Match on IP and Method Policy".

https://docs.trafficserver.apache.org/en/10.0.x/admin-guide/files/remap.config.en.html#match-on-ip-and-method-policy

Reply:
Sounds good. I'll revert to current names then tomorrow.

Reply:
Thank you!

Reply:
I'll replace this PR with one that does not do the name changes:
https://github.com/apache/trafficserver/pull/11633